File: educational-core/src/com/jetbrains/edu/learning/serialization/converter/json/ToFourthVersionJsonStepOptionsConverter.java
package com.jetbrains.edu.learning.serialization.converter.json;

import com.google.gson.JsonObject;
import com.google.gson.JsonPrimitive;
import com.jetbrains.edu.learning.EduNames;
import com.jetbrains.edu.learning.stepik.StepikNames;
import org.jetbrains.annotations.NotNull;

import static com.jetbrains.edu.learning.serialization.SerializationUtils.Json.TITLE;

public class ToFourthVersionJsonStepOptionsConverter implements JsonStepOptionsConverter {

  @NotNull
  @Override
  public JsonObject convert(@NotNull JsonObject stepOptionsJson) {
    if (stepOptionsJson.has(TITLE) &&
        StepikNames.PYCHARM_ADDITIONAL.equals(stepOptionsJson.get(TITLE).getAsString())) {
      stepOptionsJson.remove(TITLE);
      stepOptionsJson.add(TITLE, new JsonPrimitive(EduNames.ADDITIONAL_MATERIALS));
    }
    return stepOptionsJson;
  }
}

File: educational-core/src/com/jetbrains/edu/learning/serialization/converter/json/ToFourthVersionJsonStepOptionsConverter.kt
package com.jetbrains.edu.learning.serialization.converter.json;

import com.fasterxml.jackson.databind.node.ObjectNode;
import com.jetbrains.edu.learning.EduNames;
import com.jetbrains.edu.learning.stepik.StepikNames;
import org.jetbrains.annotations.NotNull;

import static com.jetbrains.edu.learning.serialization.SerializationUtils.Json.TITLE;

public class ToFourthVersionJsonStepOptionsConverter implements JsonStepOptionsConverter {

  @NotNull
  @Override
  public ObjectNode convert(@NotNull ObjectNode stepOptionsJson) {
    if (stepOptionsJson.has(TITLE) &&
        StepikNames.PYCHARM_ADDITIONAL.equals(stepOptionsJson.get(TITLE).asText())) {
      stepOptionsJson.remove(TITLE);
      stepOptionsJson.put(TITLE, EduNames.ADDITIONAL_MATERIALS);
    }
    return stepOptionsJson;
  }
}
